system/system.proto

```diff
@@ -108,7 +108,8 @@ message RebootRequest {
   uint64 delay = 2;
   // Informational reason for the reboot.
   string message = 3;
-  // Optional sub-components to reboot.
+  // Optional, sub-components to reboot.
+  // If it's not set, reboot the whole network device.
```

Comment:
hmm i feel like we should require the component to reboot since rebooting a chassis as the default seems like a rather large hammer to "default" to

Reply:
This PR does require the user/client to be careful with the big hammer. 

I had the similar concern, but I also debated with myself that if the field is required, then we probably want to have a very standard way (across all the vendors) to say this `Foo` component/sub-component means the whole device for all vendors? Is it safe to use ` /components/component/name (identity=CHASSIS)` here? @dplore

Reply:
If we use some vendor into consideration, the chassis might have the following component - CPM that is equipped in a given slot (in a redundant CPM mode). Which is modelled as `/platform/control[slot=*]`.
So CPM_A would be `/platform/control[slot=A]` and CPM_B would be at `/platform/control[slot=B]`

1. What would be the component path in OC notation? Will the individual CPMs in their respective slots be modelled as subcomponents or individual components?
2. in any case, if we put a say `component[name=control-plane-module]` in a request and do not specify any potential subcomponents, then rebooting the whole network device seems like an overkill. Should it rather be a reboot of all subcomponents instead?

Reply:
Should the `subcomponent` be renamed to `component`? What is the logic behind naming it a `subcomponent`? In openconfig-platform we have `component -> subcomponent` relationship. If we assume that reboot works on paths defined in openconfig-platform, then maybe using a subcomponent name is not entirely logical.

Reply:
yes component makes sense - and if not set return not found or invalid parameter

Reply:
then I suppose to reboot the chassis users should use the `/components/component[name=chassis]` or something similar to indicate the the control plane modules should reboot?

Reply:
yes

Reply:
We should require a single component.  If the component is not specified, there should be an error.  

Also agreed that we should explicitly clarify that reboot of /components/component[name=chassis]  indicates a full system reboot.